File: example-compile-profile/compile/helloMPIfortran/helloMPI.f

       



       program hello
       include 'mpif.h'

       
       integer numtasks, rank, len, ierr  
       character(MPI_MAX_PROCESSOR_NAME) hostname
     
       call MPI_INIT(ierr)
       if (ierr .ne. MPI_SUCCESS) then
          print *,'Error starting MPI program. Terminating.'
          call MPI_ABORT(MPI_COMM_WORLD, rc, ierr)
       end if
     
       call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)
       call MPI_COMM_SIZE(MPI_COMM_WORLD, numtasks, ierr)
       call MPI_GET_PROCESSOR_NAME(hostname, len, ierr)
       print *, 'Hello from process ',rank,' of ', numtasks,'.'
       print *, ' I am running on host: ',hostname, '.'
       
       print *, ' SQRT(rank): ',SQRT(rank+3.3)
 
       call MPI_FINALIZE(ierr)
       
       end

File: example-compile-profile/compile/helloMPIc/helloMPI.c
#include <stdio.h>
#include <mpi.h>

int main (int argc, char* argv[]) {
  int rank, size,res;
  char myName[64];

  MPI_Init (&argc, &argv);                  
  MPI_Comm_rank (MPI_COMM_WORLD, &rank);  
  MPI_Comm_size (MPI_COMM_WORLD, &size); 

  MPI_Get_processor_name(myName,&res);
  printf( "Hello from process %d of %d. I am running on host: %s.\n", rank, size, myName );
  
  MPI_Finalize();
  
  return 0;
}
